runtime/sam/expr/function/time.go

```diff
@@ -46,3 +49,34 @@ func (b *Bucket) Call(ectx expr.Context, args []zed.Value) zed.Value {
 	}
 	return zed.NewTime(nano.Ts(v).Trunc(bin))
 }
+
+// https://github.com/brimdata/zed/blob/main/docs/language/functions.md#strftime
+type Strftime struct {
+	zctx      *zed.Context
+	format    string
+	formatter *strftime.Strftime
+}
+
+func (s *Strftime) Call(ectx expr.Context, args []zed.Value) zed.Value {
+	formatArg, timeArg := args[0], args[1]
+	if !formatArg.IsString() {
+		return s.zctx.WrapError(ectx.Arena(), "strftime: string value required for format arg", formatArg)
+	}
+	if formatArg.IsNull() {
+		return zed.NullString
+	}
+	format := formatArg.AsString()
+	if zed.TypeUnder(timeArg.Type()) != zed.TypeTime {
+		return s.zctx.WrapError(ectx.Arena(), "strftime: time value required for time arg", args[1])
+	}
+	t := timeArg.AsTime()
+	if s.format != format {
```

Comment:
Nit: No need for Strftime.format.
```suggestion
	if s.formatter == nil || s.formatter.Pattern() != format {
```